polaris-migrator/src/migrations/replace-sass-motion/replace-sass-motion.ts

```diff
@@ -1,33 +1,144 @@
 import type {FileInfo, API, Options} from 'jscodeshift';
-import postcss, {Plugin, Declaration, Helpers} from 'postcss';
-import valueParser, {
-  ParsedValue,
-  Node,
-  FunctionNode,
-} from 'postcss-value-parser';
+import postcss, {Plugin} from 'postcss';
+import valueParser from 'postcss-value-parser';
 
 import {POLARIS_MIGRATOR_COMMENT} from '../../constants';
 import {
   NamespaceOptions,
   namespace,
   isSassFunction,
-  hasNumericOperator,
+  isNumericOperator,
+  getFunctionArgs,
+  isTransformableDuration,
 } from '../../utilities/sass';
 import {isKeyOf} from '../../utilities/type-guards';
 
+export default function replaceSassMotion(
+  fileInfo: FileInfo,
+  _: API,
+  options: Options,
+) {
+  return postcss(plugin(options)).process(fileInfo.source, {
+    syntax: require('postcss-scss'),
+  }).css;
+}
+
 const processed = Symbol('processed');
-const DEFAULT_DURATION = 'base';
 
-const durationFuncMap = {
-  none: '--p-duration-0',
-  fast: '--p-duration-100',
-  base: '--p-duration-200',
-  slow: '--p-duration-300',
-  slower: '--p-duration-400',
-  slowest: '--p-duration-500',
+interface PluginOptions extends Options, NamespaceOptions {}
+
+const plugin = (options: PluginOptions = {}): Plugin => {
+  const namespacedDuration = namespace('duration', options);
+
+  return {
+    postcssPlugin: 'replace-sass-motion',
+    Declaration(decl) {
+      // @ts-expect-error - Skip if processed so we don't process it again
```

Comment:
I battled TypeScript for over half a day to get rid of this comment. The `withParsedValue` higher order function was written to ensure the correct typing while also supporting the addition of the `[processed]` key, plus the pre/post processing required by `postcss-value-parser`.